core/src/main/kotlin/io/customer/sdk/data/store/GlobalPreferenceStore.kt

```diff
@@ -0,0 +1,36 @@
+package io.customer.sdk.data.store
+
+import android.content.Context
+import androidx.core.content.edit
+
+/**
+ * Store for global preferences that are not tied to a specific api key, user
+ * or any other entity.
+ */
+interface GlobalPreferenceStore {
+    fun saveDeviceToken(token: String)
+    fun getDeviceToken(): String?
+
+    fun clearAll()
+}
+
+internal class GlobalPreferenceStoreImpl(
+    context: Context
+) : PreferenceStore(context), GlobalPreferenceStore {
```

Comment:
This is same as [`SitePreferenceRepository` in tracking](https://github.com/customerio/customerio-android/blob/c8cbb12eea9502a2ad8be4560e365d194124f59d/sdk/src/main/java/io/customer/sdk/repository/preference/SitePreferenceRepository.kt) today. The naming is updated to be more flexible.